include/envoy/server/filter_config.h

```diff
@@ -151,6 +154,24 @@ class NamedNetworkFilterConfigFactory {
                                                      FactoryContext& context) PURE;
 
   /**
+   * v2 variant of createFilterFactory(..), where filter configs are specified as proto. This may be
+   * optionally implemented today, but will in the future become compulsory once v1 is deprecated.
+   */
+  virtual NetworkFilterFactoryCb createFilterFactoryFromProto(const Protobuf::Message& config,
+                                                              FactoryContext& context) {
+    UNREFERENCED_PARAMETER(config);
+    UNREFERENCED_PARAMETER(context);
+    return NetworkFilterFactoryCb();
+  }
+
+  /**
+   * @return ProtobufTypes::MessagePtr create empty config proto message for v2. The filter
+   *         config will be parsed into the result of this method. Optional today, will be
+   *         compulsory when v1 is deprecated.
+   */
+  virtual ProtobufTypes::MessagePtr createConfigProto() { return nullptr; }
```

Comment:
Per offline convo I think we decided to switch to createEmptyConfigProto() and have a slightly more robust comment about struct -> JSON -> message. Otherwise change LGTM.